src/mixins/itext_click_behavior.mixin.ts

```diff
@@ -1,31 +1,66 @@
 //@ts-nocheck
 import { ObjectEvents } from '../EventTypeDefs';
 import { IPoint, Point } from '../point.class';
+import type { DragMethods } from '../shapes/Object/InteractiveObject';
 import { TPointerEvent, TransformEvent } from '../typedefs';
 import { stopEvent } from '../util/dom_event';
 import { invertTransform, transformPoint } from '../util/misc/matrix';
+import { DraggableTextDelegate } from './DraggableTextDelegate';
 import { ITextKeyBehaviorMixin } from './itext_key_behavior.mixin';
 
-export abstract class ITextClickBehaviorMixin<
-  EventSpec extends ObjectEvents
-> extends ITextKeyBehaviorMixin<EventSpec> {
+export abstract class ITextClickBehaviorMixin<EventSpec extends ObjectEvents>
+  extends ITextKeyBehaviorMixin<EventSpec>
+  implements DragMethods
+{
+  private declare __lastSelected: boolean;
   private declare __lastClickTime: number;
   private declare __lastLastClickTime: number;
   private declare __lastPointer: IPoint | Record<string, never>;
   private declare __newClickTime: number;
 
-  /**
-   * Initializes "dbclick" event handler
-   */
-  initDoubleClickSimulation() {
-    this.__lastClickTime = +new Date();
+  protected draggableTextDelegate: DraggableTextDelegate;
+
+  initBehavior() {
+    // Initializes event handlers related to cursor or selection
+    this.initMousedownHandler();
+    this.initMouseupHandler();
+    this.initClicks();
 
+    // Initializes "dbclick" event handler
+    this.__lastClickTime = +new Date();
     // for triple click
     this.__lastLastClickTime = +new Date();
-
     this.__lastPointer = {};
-
     this.on('mousedown', this.onMouseDown);
+
+    // TODO: replace this with a standard assignment when shitty `clone` is removed
+    Object.defineProperty(this, 'draggableTextDelegate', {
+      value: new DraggableTextDelegate(this),
+      configurable: false,
+      enumerable: false,
+      writable: true,
+    });
+
+    super.initBehavior();
+  }
+
+  shouldStartDragging() {
+    return this.draggableTextDelegate.isActive();
+  }
+
+  /**
+   * @public override this method to control whether instance should/shouldn't become a drag source, @see also {@link DraggableTextDelegate#isActive}
+   * @returns {boolean} should handle event
+   */
+  onDragStart(e: DragEvent) {
+    return this.draggableTextDelegate.onDragStart(e);
+  }
+
+  /**
+   * @public override this method to control whether instance should/shouldn't become a drop target
+   */
+  canDrop(e: DragEvent) {
+    return this.draggableTextDelegate.canDrop(e);
```

Comment:
why we would suggest to overrid the class and not create a custom draggable test delegate?

Reply:
both can be done...
I was thinking of your guideline of being able to control dnd with ease, so I felt it is best to have methods on the instance so the dev doesn't need to understand the delegate